src/main/java/ua/hillel/moneymanager/service/category/CategoryServiceImpl.java

```diff
@@ -0,0 +1,83 @@
+package ua.hillel.moneymanager.service.category;
+
+import jakarta.transaction.Transactional;
+import lombok.AllArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Component;
+import ua.hillel.moneymanager.dto.CategoryDto;
+import ua.hillel.moneymanager.entity.Category;
+import ua.hillel.moneymanager.mapper.CategoryMapper;
+import ua.hillel.moneymanager.repository.CategoryRepository;
+import ua.hillel.moneymanager.repository.UserRepository;
+
+import java.util.List;
+import java.util.Objects;
+
+@Component
+@AllArgsConstructor
+public class CategoryServiceImpl implements CategoryService {
```

Comment:
A class should be transactional, meaning that you should put the annotation `@Transactional` on the class level. All read methods - should have `@Transactional(readOnly = true)`